app/tasks/research_tasks.py

```
@@ -691,8 +762,376 @@ def _generate_placeholder_report(job: dict, artifacts: list) -> str:
+        pdf_bytes = buffer.getvalue()
+        buffer.close()
+
+        logger.info(
+            "PDF generated",
+            size_bytes=len(pdf_bytes)
+        )
+
+        return pdf_bytes
+
+    except ImportError as e:
+        logger.error(f"ReportLab not available: {e}")
```

Ruff (0.14.13)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)